server/mergin/sync/public_api_controller.py

```diff
@@ -177,12 +178,15 @@ def add_project(namespace):  # noqa: E501
 
     :rtype: None
     """
-    request.json["name"] = request.json["name"].strip()
+    name = request.json["name"].strip()
 
-    if not is_name_allowed(request.json["name"]):
+    if not name:
+        abort(400, "Project name cannot be empty")
+    validation = project_name_validation(name)
```

Comment:
maybe rename to `validation_error` / `error` / `is_valid` to be clearer what an output is